src/tiles/tile.ts

```diff
@@ -1,17 +1,20 @@
-import type { TileMatrixId } from '../types';
-import { validateMetatile } from '../validations/validations';
+import type { BoundingBox, GeoPoint } from '../classes';
+import type { ArrayElement } from '../types';
+import { validateMetatile, validateTileByTileMatrix, validateTileMatrix } from '../validations/validations';
 import type { TileMatrixSet } from './tileMatrixSet';
+import { TileRange } from './tileRange';
+import { geoCoordsToTile, tileToGeoCoords } from './tiles';
+import type { TileMatrixId } from './types';
 
 /**
  * Tile class that supports a metatile definition
  */
-
 export class Tile<T extends TileMatrixSet> {
   public constructor(
-    public col: number,
-    public row: number,
-    public tileMatrixId: TileMatrixId<T>,
-    public metatile?: number
+    public readonly col: number,
+    public readonly row: number,
+    public readonly tileMatrixId: TileMatrixId<T>,
```

Comment:
we should somehow validate the tile against some form of tile matrix
a tile without a tile matrix context doesn't have any spatial context
and so isn't very useful

currently, `tileMatrixId` is not enough to valiate because matrix width/height
are unavailable at validation time
by changing the class to hold a tile matrix def it could be validated against it
but by doing so we will tightly couple `tile` and `tileMatrix`